spec/src/main/resources/META-INF/smithy/bsp/bsp.smithy

```diff
@@ -614,6 +614,9 @@ structure BuildServerCapabilities {
     /// necessary information required to launch a Java process for testing or
     /// debugging.
     jvmTestEnvironmentProvider: Boolean = false
+    /// The server can respond to `workspace/cargoFeaturesState` and
+    /// `setCargoFeatures` requests. In other words, supports Cargo Features extension.
+    cargoFeaturesProvider: Boolean = false
```

Comment:
It seems this `BuildServerCapabilities` is dependent on all extensions. In the future it may be worth splitting into many extension capabilities. It could look like this:

```smithy
structure BuildServerCapabilities {
  /// common providers

  extensionCapabilities: Map[String, ExtensionCapabilities]
}
```

Reply:
Agree. Moved your comment to #609